### PR TITLE
game connection mvp

### LINKED ISSUE #14: [EBS] Private endpoint to force redeem to run
Can be useful if someone's redeem is eaten up

### PR DESCRIPTION
**Pull request by Govorunb**
closes #14 somewhat

### CODE REVIEW

File: ebs/package.json [hunk 1]
```diff
@@ -8,10 +8,12 @@
     },
     "dependencies": {
         "@twurple/ebs-helper": "^7.1.0",
+        "@types/express-ws": "^3.0.4",
```
Comment (Alexejhero) [inline ebs/package.json]: this should be installed with -D

File: ebs/src/index.ts [hunk 2]
```diff
@@ -33,18 +34,25 @@ async function main() {
             });
             break;
         } catch {
+            if (!process.env.DB_HOST) {
```
Comment (Alexejhero) [inline ebs/src/index.ts]: renamed to MYSQL_HOST

File: ebs/src/modules/game.ts
Comment (Alexejhero) [inline ebs/src/modules/game.ts:1]: move to modules/game/index.ts at the bare minimum

File: ebs/src/modules/game/connection.ts [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+    public isConnected() {
+        return this.socket?.readyState == ServerWS.OPEN;
+    }
+    public setSocket(ws: ServerWS) {
+        if (this.isConnected()) {
+            this.socket!.close();
+        }
+        this.socket = ws;
+        ws.on('connection', () => {
+            console.log(`Connected`);
```
Comment (Alexejhero) [inline ebs/src/modules/game/connection.ts]: connected to what?

File: ebs/src/modules/game/connection.ts [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+            console.log(`Connection error ${error}`);
+        })
+    }
+    public processMessage(msg: GameMessage) {
+        switch (msg.messageType) {
+            case MessageType.Hello:
+                this.handshake = true;
+                const reply = {
+                    ...this.makeMessage(MessageType.HelloBack),
+                    allowed: (msg as HelloMessage).version == VERSION,
```
Comment (Alexejhero) [inline ebs/src/modules/game/connection.ts]: this unnecessary can be fixed with proper typescript

File: ebs/src/modules/game/messages.server.ts [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+}
+
+export type ConsoleInputMessage = ServerMessage & {
+    messageType: MessageType.ConsoleInput,
+    input: string
+}
+
+export enum CommandInvocationSource {
+    Swarm,
+    Dev
```
Comment (Alexejhero) [inline ebs/src/modules/game/messages.server.ts:19]: rename to Backseat

File: ebs/src/modules/game/messages.ts [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+    timestamp: UnixTimestampUtc
+}
+
+export type TwitchUser = {
+    /** Channel id */
+    id: number,
+    /** Twitch username (login name) */
+    userName?: string,
+    /** User's chosen display name. */
+    displayName?: string
```
Comment (Alexejhero) [inline ebs/src/modules/game/messages.ts]: This should never be null, right?

**Review by Alexejhero — DISMISSED** [review]

File: ebs/src/modules/game/messages.server.ts (hunk 5 above)
Comment (Govorunb) [reply ebs/src/modules/game/messages.server.ts:19]: no

File: ebs/src/modules/game/messages.server.ts [hunk 7]
```diff
@@ -0,0 +1,27 @@
+import { MessageType, Message, TwitchUser } from "./messages";
+
+export type ServerMessage = Message & {
+    /** User that triggered the message. e.g. for redeems, the user who bought the redeem. */
+    user?: TwitchUser | null
```
Comment (Alexejhero) [inline ebs/src/modules/game/messages.server.ts]: Pick one, not both

```suggestion
    user: TwitchUser | null
```

```suggestion
    user?: TwitchUser
```

**Pull request merged**